src/display/annotation_storage.js

```diff
@@ -19,6 +19,9 @@
 class AnnotationStorage {
   constructor() {
     this._storage = new Map();
+    this._modified = false;
+    this.onModified = null;
```

Comment:
Could you add a comment for why these callbacks are necessary? If I understood the discussion correctly it's only there to ensure that we bind on `beforeunload` when forms are edited and not in other cases for performance/usability reasons, right? In that case, perhaps something like this would work?

```javascript
this._modified = false;

// Callbacks to signal when the modification state is set or reset.
// This is used by the viewer to only bind on `beforeunload` if forms
// are actually edited to prevent doing so unconditionally since that
// can have undesirable efffects.
this.onModified = null;
this.onReset = null;
```

Moreover, I would rename the callbacks to `this.onSetMofied` and `this.onResetModified` to match the method names.